### PR TITLE
[4.0] Add prepared statements for mod_users_latest

### PR DESCRIPTION
**Pull request by HLeithner**
### Summary of Changes
Updated SQL queries to prepared statements and made some cleanups around the queries.

### Testing Instructions
Use the module in all ways you can think of.

* create a frontend module
* set the filter by group
* log into frontend with a non admin user
* You should see all registration with the access level your user have access to.

### Expected result
Nothing changed.

### CODE REVIEW

File: modules/mod_users_latest/Helper/UsersLatestHelper.php [hunk 1]
```diff
@@ -50,10 +50,9 @@ public static function getUsers($params)
 
 			$query->join('LEFT', '#__user_usergroup_map AS m ON m.user_id = a.id')
 				->join('LEFT', '#__usergroups AS ug ON ug.id = m.group_id')
-				->where('ug.id in (' . implode(',', $groups) . ')')
-				->where('ug.id <> 1');
+				->whereIn($db->quoteName('ug.id'),  $groups)
```
Comment (Quy) [inline modules/mod_users_latest/Helper/UsersLatestHelper.php]:
```suggestion
				->whereIn($db->quoteName('ug.id'), $groups)
```

File: modules/mod_users_latest/Helper/UsersLatestHelper.php [hunk 2]
```diff
@@ -50,10 +50,9 @@ public static function getUsers($params)
 
 			$query->join('LEFT', '#__user_usergroup_map AS m ON m.user_id = a.id')
```
Comment (Quy) [inline modules/mod_users_latest/Helper/UsersLatestHelper.php]: Apply `$db->quoteName`

File: modules/mod_users_latest/Helper/UsersLatestHelper.php [hunk 3]
```diff
@@ -48,12 +48,11 @@ public static function getUsers($params)
 				return array();
 			}
 
-			$query->join('LEFT', '#__user_usergroup_map AS m ON m.user_id = a.id')
-				->join('LEFT', '#__usergroups AS ug ON ug.id = m.group_id')
-				->where('ug.id in (' . implode(',', $groups) . ')')
-				->where('ug.id <> 1');
+			$query->leftJoin($db->quoteName('#__user_usergroup_map', 'm') . ' ON ' . $db->quoteName('m.user_id') . ' = ' . $db->quoteName('a.id'))
```
Comment (Quy) [inline modules/mod_users_latest/Helper/UsersLatestHelper.php]:
```suggestion
			$query->leftJoin($db->quoteName('#__user_usergroup_map', 'm'), $db->quoteName('m.user_id') . ' = ' . $db->quoteName('a.id'))
```

File: modules/mod_users_latest/Helper/UsersLatestHelper.php [hunk 4]
```diff
@@ -48,12 +48,11 @@ public static function getUsers($params)
...
+				->leftJoin($db->quoteName('#__usergroups', 'ug') . ' ON ' . $db->quoteName('ug.id') . ' = ' . $db->quoteName('m.group_id'))
```
Comment (Quy) [inline modules/mod_users_latest/Helper/UsersLatestHelper.php]:
```suggestion
				->leftJoin($db->quoteName('#__usergroups', 'ug'), $db->quoteName('ug.id') . ' = ' . $db->quoteName('m.group_id'))
```

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 7bfe2eb45f6687dfbf644803eb341c1e0b539670<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25045">issues.joomla.org/tracker/joomla-cms/25045</a>.</sub>

**Comment by alikon** [comment]
I have tested this item :white_check_mark: successfully on 7bfe2eb45f6687dfbf644803eb341c1e0b539670<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25045">issues.joomla.org/tracker/joomla-cms/25045</a>.</sub>

**Comment by alikon** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25045">issues.joomla.org/tracker/joomla-cms/25045</a>.</sub>

**Comment by roland-d** [comment]
Thank you.

**Pull request merged**